### PR TITLE
Capture errors when building and running fuzz targets

### PR DESCRIPTION
**Pull request by DonggeLiu**
This PR fixes a `libfuzzer` parser error and captures other errors when building and running fuzz targets.

Related:
https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-05-15-273-dg-273-comparison/benchmark/output-libusb-libusb_get_string_descriptor_ascii/index.html

https://pantheon.corp.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22oss-fuzz%22%0Aresource.labels.location%3D%22us-central1-c%22%0Aresource.labels.cluster_name%3D%22llm-experiment%22%0Aresource.labels.namespace_name%3D%22default%22%0Alabels.k8s-pod%2Fbatch_kubernetes_io%2Fcontroller-uid%3D%22cb8a5fcb-948d-4faf-b1b7-bd16baee7f83%22%20severity%3E%3DDEFAULT%0A;cursorTimestamp=2024-05-15T07:34:10.709184162Z;startTime=2024-05-15T07:14:00.000Z?e=-13802955&mods=logs_tg_prod&project=oss-fuzz

https://pantheon.corp.google.com/cloud-build/builds;region=us-central1/a110773c-213b-47ed-9f66-eda54f0266b5;step=7?e=-13802955&mods=logs_tg_prod&project=oss-fuzz

### CODE REVIEW

File: experiment/builder_runner.py [hunk 1]
```diff
@@ -304,6 +306,21 @@ def build_and_run(self, generated_project: str, target_path: str,
...
       return build_result, None
 
+    try:
+      return self.build_and_run_local(generated_project, target_path, iteration,
+                                      build_result)
+    except Exception as err:
+      logging.warning(
+          'Error occurred when building and running fuzz target locally'
+          '(attempt %d): %s', iteration, err)
+      traceback.print_exc()
```
Comment (oliverchang) [inline experiment/builder_runner.py:316]: Should we use `format_exc()` instead and use `logging` for this?

File: experiment/builder_runner.py [hunk 2]
```diff
@@ -591,10 +608,26 @@ def _run_with_retry_control(target_path: str, *args, **kwargs) -> bool:
...
       return build_result, None
 
+    try:
+      return self.build_and_run_cloud(generated_project, target_path, iteration,
+                                      build_result)
+    except Exception as err:
+      logging.warning(
+          'Error occurred when building and running fuzz target on cloud'
+          '(attempt %d): %s', iteration, err)
+      traceback.print_exc()
```
Comment (oliverchang) [inline experiment/builder_runner.py:624]: Should we use `format_exc()` instead and use `logging` for this?

**Review by oliverchang — APPROVED** [review]

File: experiment/builder_runner.py (hunk 2 above)
Comment (DonggeLiu) [reply experiment/builder_runner.py:623]: Done!

File: experiment/builder_runner.py (hunk 1 above)
Comment (DonggeLiu) [reply experiment/builder_runner.py]: Done!

**Pull request merged**